selene-lib/src/rules/mismatched_arg_count.rs

```diff
@@ -0,0 +1,425 @@
+                    let identifier = range(name_token);
+                    let variable = self
+                        .scope_manager
+                        .variables
+                        .iter()
+                        .find(|v| v.1.identifiers.contains(&identifier));
+
+                    if let Some((id, _)) = variable {
+                        self.definitions
+                            .insert(id, ParameterCount::from_function_body(&function_body));
+                    }
+                }
+            }
+        }
+    }
+
+    fn visit_assignment(&mut self, assignment: &ast::Assignment) {
+        let mut expressions = assignment.expr_list().iter();
+
+        for var in assignment.var_list() {
+            let expression = expressions.next();
+
+            if let Some(ast::Expression::Value { value, .. }) = expression {
+                if let ast::Value::Function((_, function_body)) = &**value {
+                    let identifier = range(var);
+                    if let Some((_, reference)) = self
+                        .scope_manager
+                        .references
+                        .iter()
+                        .find(|r| r.1.identifier == identifier)
+                    {
+                        if let Some(variable) = reference.resolved {
+                            // Ignore blacklisted variables
+                            // Check to see we aren't reassigning an already assigned variable
+                            // If we are - bail out of this and blacklist it, as we don't know which function definition is correct
+                            if self.blacklisted_variables.contains(&variable) {
+                                return;
+                            } else if self.definitions.contains_key(&variable) {
+                                self.definitions.remove(&variable);
+                                self.blacklisted_variables.insert(variable);
+                            } else {
+                                self.definitions.insert(
+                                    variable,
+                                    ParameterCount::from_function_body(&function_body),
+                                );
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+struct MismatchedArgCountVisitor {
+    mismatched_arg_counts: Vec<MismatchedArgCount>,
+    scope_manager: ScopeManager,
+    definitions: HashMap<Id<Variable>, ParameterCount>,
+}
+
+impl Visitor<'_> for MismatchedArgCountVisitor {
+    fn visit_function_call(&mut self, call: &ast::FunctionCall) {
+        if_chain::if_chain! {
+            // Check that we're using a named function call, with an anonymous call suffix
+            if let ast::Prefix::Name(name) = call.prefix();
+            if let ast::Suffix::Call(ast::Call::AnonymousCall(args)) = call.iter_suffixes().next().unwrap();
```

Comment:
```suggestion
            if let Some(ast::Suffix::Call(ast::Call::AnonymousCall(args))) = call.iter_suffixes().next();
```

I know this shouldn't be possible, but if we can so easily avoid a panic here I think we should.